File: Part 4 - Clustering/Section 25 - Hierarchical Clustering/my_model.py
# Importing the libraries

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# importing data
dataset = pd.read_csv('Mall_Customers.csv')
X = dataset[['Annual Income (k$)','Spending Score (1-100)']]

# using dendrogram
import scipy.cluster.hierarchy as sch
dendrogram = sch.dendrogram(sch.linkage(X, method='ward'))

# visualizing
plt.title('dendrogram')
plt.xlabel('customers')
plt.ylabel('euclidian distance')
plt.show()

# fitting hierarchical clustering to the mall data
from sklearn.cluster import AgglomerativeClustering
hc = AgglomerativeClustering(n_clusters=5, linkage='ward')
y_hc = hc.fit_predict(X)

# visualizing the clusters
plt.scatter(X['Annual Income (k$)'], X['Spending Score (1-100)'], c=y_hc)
plt.show()

File: Part 4 - Clustering/Section 25 - Hierarchical Clustering/my_model.R
dataset = read.csv('Mall_Customers.csv')
X = dataset[4:5]

# using dendogram
dendrogram = hclust(dist(X, method = 'euclidean'), method = 'ward.D')
plot(dendrogram,
     main = paste('dendrogram'),
     xlab = 'customers',
     ylab = 'euclidean distance')

# fitting hierarchical cluster
hc = hclust(dist(X, method = 'euclidean'), method = 'ward.D')
y_hc = cutree(hc, 5)

# visualize the clusters
# Visualising the clusters
library(cluster)
clusplot(dataset,
         y_hc,
         lines = 0,
         shade = TRUE,
         color = TRUE,
         labels= 2,
         plotchar = FALSE,
         span = TRUE,
         main = paste('Clusters of customers'),
         xlab = 'Annual Income',
         ylab = 'Spending Score')